src/lib.rs

```diff
@@ -8,5 +8,10 @@ pub mod client;
 mod common;
 pub mod rt;
 pub mod server;
+#[cfg(all(
+    any(feature = "http1", feature = "http2"),
+    any(feature = "server", feature = "client")
+))]
```

Comment:
Perhaps hyper should have a `service` feature so we don't need to do stuff like this.